ouroboros-consensus-test/bench-mempool/Main.hs

```diff
@@ -117,20 +150,131 @@ main = do
   Adding TestBlock transactions to a mempool
 -------------------------------------------------------------------------------}
 
-openMempoolWithCapacityFor :: [MempoolCmd TestBlock] ->  IO (MempoolWithMockedLedgerItf IO TestBlock)
-openMempoolWithCapacityFor cmds =
+openMempoolWithCapacityFor ::
+     InitialMempoolAndModelParams IO TestBlock
+  -> [MempoolCmd TestBlock]
+  -> IO (MempoolWithMockedLedgerItf IO TestBlock)
+openMempoolWithCapacityFor params cmds =
     openMempoolWithMockedLedgerItf capacityRequiredByCmds
                                    Tracer.nullTracer
                                    TestBlock.txSize
-                                   TestBlock.sampleMempoolAndModelParams
+                                   params
   where
     capacityRequiredByCmds = Mempool.mkCapacityBytesOverride  totalTxsSize
       where totalTxsSize = sum $ fmap TestBlock.txSize $ txsAddedInCmds cmds
 
-mkNTryAddTxs :: Int -> [MempoolCmd TestBlock.TestBlock]
-mkNTryAddTxs 0 = []
-mkNTryAddTxs n =        [TryAdd [TestBlock.mkTx [] [TestBlock.Token 0]]]
-                <> fmap (TryAdd . (:[]) . mkSimpleTx) (zip [0 .. n - 2] [1 .. n - 1])
-  where
-    mkSimpleTx (x, y) = TestBlock.mkTx [TestBlock.Token (fromIntegral x)]
-                                       [TestBlock.Token (fromIntegral y)]
+{-------------------------------------------------------------------------------
+  Scenarios
+-------------------------------------------------------------------------------}
+
+-- | Scenario 1: The backing store is empty. The changelog is empty. There are
+-- @n@ linked mempool transactions. i.e., each transaction produces a token
+-- that the /next/ transaction consumes.
+scenario1 ::
+     Int
+  -> ( InitialMempoolAndModelParams m TestBlock
+     , [MempoolCmd TestBlock.TestBlock]
+     )
+scenario1 n =
+  let
+    lst   = ledgerStateFromTokens []
+    blks  = []
+    cmds  = let gtxs = mkSimpleGenesisGenTx 0
+                     : [ mkSimpleGenTx x y
+                       | (x, y) <- zip [0 .. n - 2] [1 .. n - 1]
+                       ]
+            in  fmap mkSimpleTryAdd gtxs
+  in
+    (mkParams lst blks defaultLedgerDbCfg defaultInMemoryBSS, cmds)
+
+-- | Scenario 2: The backing store is empty. The changelog has @m@ linked (see
+-- 'scenario1') entries. There are @n@ linked mempool transactions. The last
```

Comment:
Mempool transactions might be a bit confusing because these transactions are not in the mempool yet. Maybe we can say something like "candidate transactions".

Reply:
Good point